Content.Server/Teleportation/Components/DimensionPotComponent.cs

```diff
@@ -0,0 +1,62 @@
+using Content.Server.Teleportation.Systems;
+using Robust.Shared.Audio;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Teleportation.Components;
+
+/// <summary>
+/// Creates a pocket dimension map on spawn.
+/// When activated by alt verb, spawns a portal to this dimension or closes it.
+/// </summary>
+[RegisterComponent]
+[Access(typeof(DimensionPotSystem))]
+public sealed class DimensionPotComponent : Component
+{
+    /// <summary>
+    /// The portal on the pot, if it is open right now.
+    /// </summary>
+    [ViewVariables]
+    public EntityUid? PotPortal = null;
+
+    /// <summary>
+    /// The portal in the pocket dimension, usually exists.
+    /// </summary>
+    [ViewVariables]
+    public EntityUid? DimensionPortal = null;
+
+    /// <summary>
+    /// Map of the pocket dimension, usually exists.
+    /// </summary>
+    [ViewVariables]
+    public MapId PocketDimensionMap = MapId.Nullspace;
```

Comment:
Also now that I think about it we're gonna have to watch this because if we get more than a couple of these a round need to look at doing something like tabletop because map-spam would be bad.

Reply:
the whole point is to have a unique map, is there much slowdown if a bozo gets 4 of them in a super surplus

Reply:
It makes it a pain in the ass for admins (we already have like 10 maps and it's extremely annoying) and also some stuff is O(n) for map iteration like PVS.

Reply:
would pvs be better if it was a single map being loadgrid'd with the pot interior thing